client/src/components/UploadCodeDialog.js

```diff
@@ -0,0 +1,311 @@
+    border: "1 solid grey",
+    borderRadius: 5,
+    backgroundColor: "black",
+    color: "white",
+    padding: "1rem",
+  },
+  input: {
+    width: "100%",
+    lineHeight: 2,
+    marginBottom: "1rem",
+    fontWeight: 600,
+  },
+  select: {
+    border: 0,
+    "&:focus": {
+      backgroundColor: "transparent",
+    },
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: "#fff",
+  },
+}));
+
+const UploadCodeDialog = () => {
+  // languages list
+  const mode_list = useMemo(
+    () =>
+      modes.map((option) => (
+        <option key={option.value} value={option.value}>
+          {option.label}
+        </option>
+      )),
+    []
+  );
+  const classes = useStyles();
+
+  // Modal/Dailog states
+  const [open, setOpen] = React.useState(false);
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setForm({
+      title: "",
+      code: "",
+    });
+    setOpen(false);
+  };
+
+  // backdrop state
+  const [backdrop, setBackdrop] = React.useState(false);
+
+  // Snackbar state
+  const [snackbar, setSnackbar] = useState({
+    open: false,
+    error: false,
+    message: "",
+  });
+
+  // form states and controlled inputs
+  const [form, setForm] = React.useState({
+    title: "",
+    code: "",
+    mode: "jsx",
+  });
+
+  const { title, code, mode } = form; // destructuring the form object for easy use.
+
+  // handle title and language select
+  const handleCodeForm = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    setForm((prev) => {
+      return {
+        ...prev,
+        [name]: value,
+      };
+    });
+  };
+
+  // handle editor input only
+  const handleEditor = (e) => {
+    const value = e;
+    setForm((prev) => {
+      return {
+        ...prev,
+        code: value,
+      };
+    });
+  };
+
+  // Onsubmit handler
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (title === "" || code === "") {
+      setSnackbar({
+        open: true,
+        message: "Please enter all fields.",
+        error: true,
+      });
+    } else {
+      setBackdrop(true);
+      // Some API calls to server to upload code and comment.
+      // simulating a 2 sec delay for loading
+      setTimeout(() => {
+        setBackdrop(false);
+        handleClose();
+        alert(`code submitted`);
+      }, 2000);
+    }
+  };
+
+  return (
+    <div>
+      <Button
+        classes={{
+          root: classes.buttonRoot,
+          label: classes.buttonLabel,
+        }}
+        onClick={handleClickOpen}
+      >
+        Upload Code
+      </Button>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        fullWidth={true}
+        maxWidth="sm"
+        disableBackdropClick
+        disableEscapeKeyDown
+        PaperProps={{
+          classes: {
+            root: classes.paper,
+          },
+        }}
+      >
+        <div className={classes.closeButton} onClick={handleClose}>
+          X
```

Comment:
I'd prefer we use Material-UI's icons and icon buttons for this

Reply:
Sure..I will do that right away and then do the merge. Thanks